static/src/javascripts/projects/common/modules/discussion/comment-count.js

```diff
@@ -0,0 +1,141 @@
+                    <span class="commentcount2__value tone-colour js_commentcount_actualvalue">${count}</span>
+                </a>`;
+    }
+
+    if (type === 'contentImmersive') {
+        return `<a href="${url}" data-link-name="Comment count" class="commentcount2 tone-colour">
+                    ${icon}<span class="commentcount__value">${count}</span> Comments
+                </a>`;
+    }
+
+    return `<a class="fc-trail__count fc-trail__count--commentcount" href="${url}" data-link-name="Comment count">${icon} ${count}</a>`;
+};
+
+const getElementsIndexedById = (context: HTMLElement): Promise<any> =>
+    fastdom
+        .read(() => context.querySelectorAll(`[${ATTRIBUTE_NAME}]`))
+        .then(elements => {
+            if (elements.length === 0) {
+                return Promise.resolve();
+            }
+
+            return [
+                ...elements,
+            ].reduce((groupedVals: Object, el: HTMLElement): Object => {
+                const attrVal = el.getAttribute(ATTRIBUTE_NAME);
+
+                if (!groupedVals[attrVal]) {
+                    groupedVals[attrVal] = [];
+                }
+
+                groupedVals[attrVal].push(el);
+
+                return groupedVals;
+            }, {});
+        });
+
+const getContentIds = (indexedElements: IndexedElements): string =>
+    Object.keys(indexedElements).sort().join(',');
+
+const getContentUrl = (el: HTMLElement): string => {
+    const a = el.getElementsByTagName('a')[0];
+
+    return `${a ? a.pathname : ''}#comments`;
+};
+
+const updateElement = (el: HTMLElement, count: number): Promise<void> => {
+    const url = el.getAttribute('data-discussion-url') || getContentUrl(el);
+
+    if (el.getAttribute('data-discussion-closed') === 'true' && count === 0) {
+        // Discussion is closed and had no comments, we don't want to show a comment count
+        return Promise.resolve();
+    }
+
+    const format = el.getAttribute('data-commentcount-format') || '';
+    const html = getTemplate(
+        {
+            url,
+            icon: inlineSvg('commentCount16icon', ['inline-tone-fill']),
+            count: integerCommas(count) || '',
+        },
+        format
+    );
+    const meta = el.getElementsByClassName('js-item__meta');
+    const containers = meta.length ? [...meta] : [el];
+
+    return fastdom.write(() => {
+        containers.forEach(container => {
+            container.insertAdjacentHTML('beforeend', html);
+        });
+        el.removeAttribute(ATTRIBUTE_NAME);
+        el.classList.remove('u-h');
+    });
+};
+
+const renderCounts = (
+    counts: Array<{ id: string, count: number }>,
+    indexedElements: IndexedElements
+): Promise<void> => {
+    const elementUpdates = counts.map(c =>
+        indexedElements[c.id].map(el => updateElement(el, c.count))
+    );
+
+    return Promise.all(elementUpdates).then(() => {
+        mediator.emit('modules:commentcount:loaded', counts);
```

Comment:
Let's kill this and therefore the `mediator` dependency.

Reply:
NB: mediator is also used in [`init`](https://github.com/guardian/frontend/pull/17718/files#diff-962cdfad85b64468283361c03e83dfe0R134)

Reply:
Yep, you're right this instance can now go but the dependency will need to remain for the reason @SiAdcock  mentioned.